### PR TITLE
Azure Batch Reef Runtime Configuration

### PR DESCRIPTION
**Pull request by homezcx**
Azure Batch Reef Runtime Configuration

### CODE REVIEW

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/client/AzureBatchRuntimeConfiguration.java [hunk 1]
```diff
@@ -58,17 +58,59 @@
...
       .bindNamedParameter(AzureBatchAccountName.class, AZURE_BATCH_ACCOUNT_NAME)
       .bindNamedParameter(AzureBatchAccountKey.class, AZURE_BATCH_ACCOUNT_KEY)
       .bindNamedParameter(AzureBatchPoolId.class, AZURE_BATCH_POOL_ID)
-      // Bind the Azure Batch runtime
-      .bindImplementation(JobSubmissionHandler.class, AzureBatchJobSubmissionHandler.class)
-      .bindImplementation(DriverConfigurationProvider.class, AzureBatchDriverConfigurationProvider.class)
       .build();
+
+  /**
+   * Returns a HDInsight runtime configuration from the credentials stored in the given file.
```
Comment (tyclintw) [inline lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/client/AzureBatchRuntimeConfiguration.java]: Batch

File: lang/java/reef-examples/pom.xml [hunk 2]
```diff
@@ -117,6 +117,11 @@ under the License.
             <artifactId>reef-runtime-multi</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.reef</groupId>
+            <artifactId>reef-runtime-azbatch</artifactId>
+            <version>0.17.0-SNAPSHOT</version>
```
Comment (ryumbra) [inline lang/java/reef-examples/pom.xml]: I believe there is already a dependency declared for reef-runtime-azbatch on line 44. Is this necessary? If it is, let's change version to ${project.version}

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/client/AzureBatchRuntimeConfiguration.java (hunk 1 above)
Comment (ryumbra) [inline lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/client/AzureBatchRuntimeConfiguration.java]: Let's fix this comment to replace HDInsight references with Azure Batch. Also applies to line 81.

**Review by tyclintw — APPROVED** [review]

**Comment by ryumbra** [comment]
One additional question. Do you have samples/instructions on how to generate the avro file for our testing?

File: lang/java/reef-examples/pom.xml (hunk 2 above)
Comment (homezcx) [reply lang/java/reef-examples/pom.xml]: My intelliJ refused to compile it yesterday without this. I tried cleaning my project and it works now. I will remove this.

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/client/AzureBatchRuntimeConfiguration.java (hunk 1 above)
Comment (homezcx) [reply lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/client/AzureBatchRuntimeConfiguration.java]: fixed.

Comment (homezcx) [reply lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/client/AzureBatchRuntimeConfiguration.java]: fixed.

**Pull request merged**